File: 2021/day-1/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	f, _ := os.Open(os.Args[1])
	s := bufio.NewScanner(f)
	s.Split(bufio.ScanLines)
	var input []int64
	for s.Scan() {
		i, _ := strconv.ParseInt(s.Text(), 0, 64)
		input = append(input, i)
	}
	count := 0
	for i:=0; i < len(input)-3; i++{
		win1 := input[i] + input[i+1] + input[i+2]
		win2 := input[i+1] + input[i+2] + input[i+3]
		if win2 > win1 {count++}
	}
	fmt.Printf("Count %d\n", count)
}

File: 2021/day-1/part2.py
from sys import argv

with open(argv[1], 'r') as infile:
    count = 0
    input = infile.readlines()
    for i in range(0, len(input)-3, 1):
        win1 = int(input[i]) + int(input[i+1]) + int(input[i+2])
        win2 = int(input[i+1]) + int(input[i+2]) + int(input[i+3])
    print(count)
    